### DISCUSSION TITLE
Recommended way to rebuild a projection

### DISCUSSION THREAD

**Discussion by gius** [Eventuous/eventuous#367] [Q&A]
What is the recommended way to rebuild a projection (restart subscription)?
My naive approach would be to truncate the projection table and delete the respective checkpoint. The problem is that unless a new event is stored, no event handlers are executed. Would you recommend to have a NoOp event for such scenario?

**Comment by ghandmann** [comment]
The easiest (and i think most common way) is to simply restart the projector after truncating the table and deleting the checkpoint. In this case it would just replay.

A more sophisticated solution would be to implement some WebAPI/MessageBus into the projector to trigger a replay without restarting the program.

But i don't think you should use NoOp-Events to signal purely technical needs.
